docs/dev/plugins/cloudformation/transformation/supported-types.md

```diff
@@ -0,0 +1,119 @@
+| WebApplication | Setup through CloudFormation Init on the EC2 host |
+| Nodejs | Setup through CloudFormation Init on the EC2 host |
+| JavaRuntime | No specific resource is created |
+| JavaApplication | Beanstalk Application and Environment |
+
+## NodeType Transformation Details
+
+These transformations take place in the `TransformationModelNodeVisitor`. During the [transform phase](transformation-workflow.md#transform) of the `TransformationLifeCycle`, the `visit()` of the various NodeTypes are executed in order to fill the `CloudFormationModule` with the necessary information to generate the CloudFormation template and the files needed to deploy the CloudFormation stack. The following section offers a detailed explanation of the transformation behaviour for each of these NodeTypes.
+
+### Compute
+
+Each Compute node that was previously marked in the [prepare phase](transformation-workflow.md#prepare) now gets transformed into an EC2 `Instance` resource with a corresponding `SecurityGroup` resource.
+
+#### 1. Security Group
+
+The `SecurityGroup` is created. The security group allows the opening of ports for the corresponding EC2 instance. If present, optional endpoint ports are opened on this security group.
+
+#### 2. Capability Mapping
+
+Both the OsCapability and ComputeCapablity of the Compute node are mapped to properties of the EC2 instance by the `CapabilityMapper`.
+
+First, the `CapabilityMapper` is used to figure out the ID of the Amazon Machine Image (AMI) corresponding to the OsCapability. To do this, we build an [`AmazonEC2`](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/AmazonEC2.html) client with the AWS SDK for Java to get the latest image IDs from AWS. In order to get the right image ID, a [`DescribeImagesRequest`](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/model/DescribeImagesRequest.html) is built with the properties of the OsCapability. When the image request is completed, the [`describeImages()`](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/AmazonEC2.html#describeImages-com.amazonaws.services.ec2.model.DescribeImagesRequest-) method of the EC2 client is used to get the images fulfilling the requirements of the request. From these images, the capability mapper takes the latest image ID available which in turn is used by the visitor as the `ImageId` property of the EC2 instance.
+
+Then the `CapabilityMapper` is used to figure out the right `InstanceType` for the EC2 instance corresponding to the ComputeCapability. The right instance type is determined based on the `numCpus` and `memSize` properties of the ComputeCapability. Once the right instance type has been found, it is used by the visitor as the `InstanceType` property of the EC2 instance. If no instance type fitting the requirements the ComputeCapability can be found, the transformation ends with a `TransformationFailureException`.
```

Comment:
Maybe state that the `CapabilityMapper` tries to scale up the values in order to find a `InstanceType`.